src/i_flmusic.c

```diff
@@ -146,6 +224,35 @@ static boolean I_FL_InitMusic(int device)
     }
     else
     {
+        if (device == DEFAULT_MIDI_DEVICE)
```

Comment:
I find this a bit hard to understand. What does it mean if the function is called with `device = -1`?

Reply:
This means that we initialize the sound module when the program starts with a saved device. We don't use `midi_player_menu` to store device number, we reconstruct `midi_player_menu` from `module_index` (`midi_player` in config) and `current_device`.